app/graphql/resolvers/farmer_resolver.py

```
@@ -0,0 +1,145 @@
+            try:
+                farmers = await FarmerService.search_farmers_by_location(
+                    db=db,
+                    latitude=latitude,
+                    longitude=longitude,
+                    radius_km=radius_km,
+                    skip=skip,
+                    limit=limit,
+                )
+                return [Farmer.from_model(f) for f in farmers]
+            except Exception:
+                raise StrawberryGraphQLError("Failed to search farmers by location")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)